### ISSUE TITLE
[package] qt/5.15.4: -o qt:with_dbus=True results in fatal error LNK1104: cannot open file dbus-1d.lib on Windows

### ISSUE THREAD

**Issue by tsondergaard** [conan-io/conan-center-index#11286]
### Package and Environment Details

* Package Name/Version: **qt/5.15.4** using **dbus/1.12.20**
* Operating System+version: **Windows 10 64-bit**
* Compiler+version: **Visual C++ 2019 16.11.16**
* Conan version: **conan 1.49.0**
* Python version: **Python 3.10.4**

### Conan profile

```ini
[settings]
os=Windows
os_build=Windows
arch=x86_64
arch_build=x86_64
compiler=Visual Studio
compiler.version=16
build_type=Release
[options]
[conf]
[build_requires]
[env]
```

### Steps to reproduce

In a debug build with qt:with_dbus=True, Qt looks for dbus-1d.lib, but the debug build of dbus on Windows produces dbus-1.lib without the d suffix.

To reproduce run

```
conan install qt/5.15.4@ -o qt:with_dbus=True -s build_type=Debug --build
```

Expected result:
Successful build

Actual result:
Build fails with fatal linker error:

```
linking ..\..\..\bin\qdbuscpp2xml.exe
LINK : fatal error LNK1104: cannot open file 'dbus-1d.lib'
```

**Comment by ericLemanissier** [comment]
this is a consequence of https://github.com/qt/qtbase/blob/v5.15.5-lts-lgpl/configure.json#L201. don't hesitate to make a PR which uses `tools.replace_in_file` to replace `"-ldbus-1d"` with `"-ldbus-1"` in this file.

**Comment by tsondergaard** [comment]
I have made a commit that uses tools.replace_in_file:

```diff
commit 1121773466a3330e1e7072dbc655909933b34880 (HEAD -> master)
Author: Thomas Sondergaard <thomas@sondergaard.cc>
Date:   Mon Jun 20 13:36:51 2022 +0200

    qt: Link dbus-1 not dbus-1d in Debug builds
    
    Neither the Conan or the vanilla CMake-build version of the dbus
    package use a d-suffix for the debug version of the library.
    
    Fixes #11286

diff --git a/recipes/qt/5.x.x/conanfile.py b/recipes/qt/5.x.x/conanfile.py
index 5f34ece83..c3ca32121 100644
--- a/recipes/qt/5.x.x/conanfile.py
+++ b/recipes/qt/5.x.x/conanfile.py
@@ -420,6 +420,10 @@ class QtConan(ConanFile):
                                   "  if (enable_precompiled_headers) {\n    if (is_win) {",
                                   "  if (enable_precompiled_headers) {\n    if (false) {"
                                   )
+        tools.replace_in_file(os.path.join(self.source_folder, "qt5", "qtbase", "configure.json"),
+                                  "-ldbus-1d",
+                                  "-ldbus-1"
+                                  )
 
     def _make_program(self):
         if self._is_msvc:
```

I'll test it and make a pull request as soon as I can.
